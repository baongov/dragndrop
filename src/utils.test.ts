import {
  generateColorMatrix,
  isEqual,
  getBlockColor,
  swapBlock,
  stringToPosition,
} from "./utils";

describe("generateColorMatrix", () => {
  it("return correct output", () => {
    expect(generateColorMatrix(2, [20, 20])).toMatchSnapshot();
  });
});

describe("isEqual", () => {
  it("should return true when two tuples are the same", () => {
    expect(isEqual([5, 9], [5, 9])).toMatchSnapshot();
  });
  it("should return false when two tuples are different", () => {
    expect(isEqual([5, 9], [-5, 9])).toMatchSnapshot();
  });
});

describe("getBlockColor", () => {
  it("return correct color tuple", () => {
    expect(
      getBlockColor(
        [[{ color: [1, 2, 3], position: [0, 0] }]],
        [0, 0],
        [-1, -1, -1]
      )
    ).toMatchSnapshot();
  });
});

describe("swapBlock", () => {
  it("return correct board after swapping", () => {
    expect(
      swapBlock(
        [
          [{ color: [1, 2, 3], position: [1, 1] }],
          [{ color: [4, 5, 6], position: [1, 1] }],
        ],
        [0, 0],
        [1, 0]
      )
    ).toMatchSnapshot();
  });
  it("return correct board if passing invalid position", () => {
    expect(
      swapBlock(
        [
          [{ color: [1, 2, 3], position: [1, 1] }],
          [{ color: [4, 5, 6], position: [1, 1] }],
        ],
        [0, 0],
        [-1, -1]
      )
    ).toMatchSnapshot();
  });
});

describe("stringToPosition", () => {
  it("return correct output", () => {
    expect(stringToPosition("2,6")).toMatchSnapshot();
    expect(stringToPosition("6")).toMatchSnapshot();
    expect(stringToPosition("6,3,5")).toMatchSnapshot();
    expect(stringToPosition(",,,23")).toMatchSnapshot();
  });
});
