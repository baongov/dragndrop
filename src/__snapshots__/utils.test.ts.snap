// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateColorMatrix return correct output 1`] = `
Array [
  Array [
    Object {
      "color": Array [
        0,
        80,
        10,
      ],
      "position": Array [
        20,
        20,
      ],
    },
    Object {
      "color": Array [
        0,
        80,
        90,
      ],
      "position": Array [
        20,
        70,
      ],
    },
  ],
  Array [
    Object {
      "color": Array [
        320,
        80,
        10,
      ],
      "position": Array [
        70,
        20,
      ],
    },
    Object {
      "color": Array [
        320,
        80,
        90,
      ],
      "position": Array [
        70,
        70,
      ],
    },
  ],
]
`;

exports[`getBlockColor return correct color tuple 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`isEqual should return false when two tuples are different 1`] = `false`;

exports[`isEqual should return true when two tuples are the same 1`] = `true`;

exports[`stringToPosition return correct output 1`] = `
Array [
  2,
  6,
]
`;

exports[`stringToPosition return correct output 2`] = `
Array [
  -1,
  -1,
]
`;

exports[`stringToPosition return correct output 3`] = `
Array [
  6,
  3,
]
`;

exports[`stringToPosition return correct output 4`] = `
Array [
  -1,
  -1,
]
`;

exports[`swapBlock return correct board after swapping 1`] = `
Array [
  Array [
    Object {
      "color": Array [
        4,
        5,
        6,
      ],
      "position": Array [
        1,
        1,
      ],
    },
  ],
  Array [
    Object {
      "color": Array [
        1,
        2,
        3,
      ],
      "position": Array [
        1,
        1,
      ],
    },
  ],
]
`;

exports[`swapBlock return correct board if passing invalid position 1`] = `
Array [
  Array [
    Object {
      "color": Array [
        1,
        2,
        3,
      ],
      "position": Array [
        1,
        1,
      ],
    },
  ],
  Array [
    Object {
      "color": Array [
        4,
        5,
        6,
      ],
      "position": Array [
        1,
        1,
      ],
    },
  ],
]
`;
